substrate/frame/state-trie-migration/README.md

```diff
@@ -0,0 +1,240 @@
+
+First, please read the documentation of this pallet. Here, we recap the steps needed to use the migration pallet. To give a brief recap, we suggest:
+
+1. Parachains use the signed migration, as it allows more control.
+2. Relay/solo chains use the automatic migration, as it is easier.
+
+Nonetheless, the safest possible option is to always use the signed migration.
+
+### 1. Adding the pallet to the runtime
+
+The following documented snippet shows the parameters that need to be decided upon in your chain in order to configure this pallet. Note that the values below are simply an example, and the exact value for each chain would vary.
+
+```rust
+use pallet_state_trie_migration::MigrationLimits;
+parameter_types! {
+    // The deposit configuration for the signed migration. 
+    // Especially if you want to allow any signed account to do the migration (see `SignedFilter`, these deposits should be high)
+    pub const MigrationSignedDepositPerItem: Balance = 1 * CENTS;
+    pub const MigrationSignedDepositBase: Balance = 20 * DOLLARS;
+}
+
+impl pallet_state_trie_migration::Config for Runtime {
+    type Event = Event;
+    type Currency = Balances;
+    type SignedDepositPerItem = MigrationSignedDepositPerItem;
+    type SignedDepositBase = MigrationSignedDepositBase;
+    // An origin that can control the whole pallet: should be Root, or a part of your council. 
+    type ControlOrigin = EnsureRoot<AccountId>;
+    // Warning: this is not advised, as it might allow the chain to be temporarily DOS-ed. 
+    // If the chain's governance/maintenance team is planning on using a specific account for the migration, 
+    // put it here to make sure only that account can trigger the signed migrations.
+    type SignedFilter = EnsureSigned<Self::AccountId>;
+    // Replace this with weight based on your runtime. 
+    type WeightInfo = ();
+}
+```
+
+You also need to include the pallet in your `contruct_runtime!`, and `define_benchmarks!`. **Don't forget to update `type WeightInfo` once you have realistic weights**!
+
+### 2.1 Running the automatic migration
+
+To run the automatic migration, you need to use the `ControlOrigin` to set 1 storage value that determines how much resources should be used per block. Once you set this, the pallet will automatically migrate enough items `on_initialize` of all the following blocks, until all keys are migrated.
+
+> `on_initialize` might go over the limits provided, because we can never know the real size of a key before reading it, at which point it is actually too late. Thus, it should NOT be used in a parachain.
+> If you are sure you want to do the migration this way, you should set `SignedFilter` such that no account can call the manual signed migrations, just in case.
+
+Let's look at `substrate-node --dev --execution Native -lruntime::state-trie-migration=debug` chain that includes this pallet. Initially, the logs don't show anything relevant.
```

Comment:
```suggestion
Let's look at `staging-node-cli --dev --execution Native -lruntime::state-trie-migration=debug` chain that includes this pallet. Initially, the logs don't show anything relevant.
```

I think we have renamed this binary now, it is in `substrate/bin/node` for you to double check.